frameworks/rooch-nursery/sources/inscribe_factory.move

```diff
@@ -0,0 +1,927 @@
+
+        (true, option::none<String>())
+    }
+    
+    fun mint_bitseed(metadata: &SimpleMap<String,vector<u8>>, seed: vector<u8>, content_type: Option<String>, body: vector<u8>) : Result<Object<Bitseed>> {
+        let is_valid_result = is_valid_bitseed(metadata);
+        if (is_err(&is_valid_result)) {
+            return as_err(is_valid_result)
+        };
+
+        let tick = get_SFT_tick(metadata);
+        let attributes = get_SFT_attributes(metadata);
+        let amount = get_SFT_amount(metadata);
+
+        if (!tick_info::is_deployed(bitseed::metaprotocol(), tick)) {
+            return err(b"the tick is not deployed")
+        };
+        let tick_info = tick_info::borrow_tick_info(bitseed::metaprotocol(), tick);
+        let has_user_input = tick_info::has_user_input(tick_info);
+        let deploy_args = option::destroy_with_default(tick_info::deploy_args(tick_info), vector::empty());
+
+        let user_input = string::utf8(b"");
+        if (has_user_input) {
+            //TODO handle user input
+            let user_input_option = get_SFT_string_attribute(&attributes, b"id");
+            if (option::is_none(&user_input_option)) {
+                return err(b"metadata.attributes.user_input is required")
+            };
+
+            user_input = *option::borrow(&user_input_option);
+        };
+        // The generator tick has no generator, so we skip the inscribe verify
+        if(tick != utf8(BIT_SEED_GENERATOR_TICK)) {
+            let generator_inscription_id_option = tick_info::generator(tick_info);
+            if (option::is_none(&generator_inscription_id_option)) {
+                return err(b"the tick can not mint on Bitcoin")
+            };
+
+            let generator_inscription_id = option::destroy_some(generator_inscription_id_option);
+            if (!ord::exists_metaprotocol_validity(generator_inscription_id)) {
+                return err(b"generator_inscription_id is not validity bitseed")
+            };
+
+            let generator_txid = ord::inscription_id_txid(&generator_inscription_id);
+            let generator_index = ord::inscription_id_index(&generator_inscription_id);
+            let inscription_obj = ord::borrow_inscription(generator_txid, generator_index);
+
+            let inscrption = object::borrow(inscription_obj);
+            let wasm_bytes = ord::body(inscrption);
+
+            let (is_valid, reason) = inscribe_verify(wasm_bytes, deploy_args, seed, user_input, metadata, content_type, body);
+            if (!is_valid) {
+                return result::err_string(option::destroy_with_default(reason, utf8(b"inscribe verify fail")))
+            };
+        };
+        let bitseed_result = tick_info::mint_on_bitcoin(bitseed::metaprotocol(), tick, amount);
+        bitseed_result
+    }
+
+    public fun inscribe_verify(wasm_bytes: vector<u8>, deploy_args: vector<u8>,
```

Comment:
The return value here is not changed to Result<bool>

Reply:
I will refactor it in a new PR.